### ISSUE TITLE
RCS1171 doesn't trigger for complex statements

### ISSUE THREAD

**Issue by IanKemp** [JosefPihrt/Roslynator#266]
Consider:

```csharp
public enum A
{
    B,
    C,
}

public class D
{
    private A? aye = null;
    public A Aye
    {
        get
        {
            if (aye == null) // RCS1171
            {
                aye = PopulateAye();
            }

            return aye.Value;
        }
    }
}
```

Refactoring Essentials correctly suggests to rewrite the above as:

```csharp
public class D
{
    private A? aye = null;
    public A Aye
    {
        get
        {
            return (aye ?? (aye = PopulateAye())).Value;
        }
    }
}
```

Now consider:

```csharp
public class E
{
    private A? aye = null;
    public A? Aye { get; set; }
}

public class F
{
    private E eee = new E();

    private A AyeEncapsulated
    {
        get
        {
            if (eee.Aye == null) // no RCS1171
            {
                eee.Aye = eee.PopulateAye();
            }

            return eee.Aye.Value;
        }
    }
}
```

RCS1171 is not suggested in this case, even though the code can be condensed to the following:

```csharp
public class E
{
    private A AyeEncapsulated
    {
        get
        {
            return (eee.Aye ?? (eee.Aye = eee.PopulateAye())).Value;
        }
    }
}
```

I'm not sure if this is a bug, or an intentional decision to not simplify complex statements (which could hurt readability/comprehension).

**Comment by JosefPihrt** [comment]
It is reasonable to make it working for expressions like `a.b`.

Expressions like `a.b.c` (and longer) could hurt readability as you pointed out.

But it is possible to save `a.b` in local variable and then apply a code fix.
